xla/service/cpu/onednn_contraction_rewriter.h

```diff
@@ -61,18 +63,26 @@ class OneDnnContractionRewriter : public HloModulePass {
   const tsl::thread::ThreadPool* compile_threadpool_;
 };
 
-#define HANDLE_OP_INTERNAL(internal_callee, contraction, ...)          \
-  switch (contraction->backend_config<BackendConfig>()                 \
-              ->backend_config_oneof_case()) {                         \
-    case BackendConfig::BackendConfigOneofCase::kOnednnMatmulConfig:   \
-      return internal_callee<                                          \
-          BackendConfig::BackendConfigOneofCase::kOnednnMatmulConfig>( \
-          contraction, __VA_ARGS__);                                   \
-    default:                                                           \
-      return internal_callee<                                          \
-          BackendConfig::BackendConfigOneofCase::kOnednnConvConfig>(   \
-          contraction, __VA_ARGS__);                                   \
+using ContractionVariant = std::variant<PrimitiveTrait<kOnednnConvConfig>,
+                                        PrimitiveTrait<kOnednnMatmulConfig>>;
+using FusionsConfigPointer = xla::cpu::OneDnnFusionConfig*;
+using OptimizationsConfigPointer = xla::cpu::OneDnnOptimizationConfig*;
```

Comment:
Per Google's [C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Aliases), we avoid making aliases in header if their purpose is only for convenience. We only have aliases in header when it's intended to be part of the API (e.g., for users to use).

I think we can keep `ContractionVariant`, but maybe rename it to `OneDnnContractionVariant` (or put it in a `onednn` namespace) to make it clear this is specific to oneDNN.

`FusionsConfigPointer` and `OptimizationsConfigPointer` seem trivial and shouldn't be declared here.

(Aliases are fine in .cc files.)

Reply:
Done